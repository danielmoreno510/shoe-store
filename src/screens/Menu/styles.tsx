import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  safeAreaContainer: {
    flex: 1,
    backgroundColor: 'white',
  },
  image: {
    width: 30,
    height: 30,
  },
  optionContainer: {
    paddingHorizontal: 16,
  },
  rowContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  option: {
    paddingVertical: 12,
  },
  textOption: {
    fontSize: 16,
    color: 'black',
    paddingHorizontal: 10,
  },
  containerNotification: {
    justifyContent: 'center',
    alignItems: 'center',
    width: 20,
    height: 20,
    borderRadius: 10,
    borderWidth: 1,
    borderColor: 'red',
    paddingLeft: 1,
    paddingTop: 1,
  },
  notification: {
    color: 'red',
    fontSize: 12,
  },
});
