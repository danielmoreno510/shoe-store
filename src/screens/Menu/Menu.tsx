import React from 'react';
import {SafeAreaView, ScrollView, TouchableOpacity, View, Text} from 'react-native';
import {Icon} from 'native-base';

import styles from './styles';
import {menuOptions} from '../../constants';

const Menu: React.FC<Navigation> = ({navigation}) => {
  const changeScreeen = (screen: Page) => () => navigation.navigate(screen);

  return (
    <SafeAreaView style={styles.safeAreaContainer}>
      <ScrollView>
        {menuOptions.map(option => (
          <View key={option.name} style={styles.optionContainer}>
            <TouchableOpacity style={styles.option} onPress={changeScreeen(option.name)}>
              <View style={styles.rowContainer}>
                <Icon type="MaterialIcons" name={option.icon} />
                <Text style={styles.textOption}>{option.name}</Text>
              </View>
            </TouchableOpacity>
          </View>
        ))}
      </ScrollView>
    </SafeAreaView>
  );
};

export default Menu;
