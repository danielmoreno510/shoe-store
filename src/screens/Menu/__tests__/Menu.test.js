import React from 'react';
import {shallow} from 'enzyme';

import Menu from '../Menu';

describe('Menu component', () => {
  describe('GIVEN a set of valid props', () => {
    const testProps = {
      navigation: {navigate: jest.fn()},
    };

    describe('WHEN the component is rendered', () => {
      const shallowWrapper = shallow(<Menu {...testProps} />);

      it('THEN should display a regular Menu', () => {
        expect(shallowWrapper).toMatchSnapshot();
      });
    });

    describe('WHEN the menu option is pressed', () => {
      const shallowWrapper = shallow(<Menu {...testProps} />);
      shallowWrapper.find('ForwardRef').at(0).simulate('press');

      it('THEN navigation function should be called', () => {
        expect(testProps.navigation.navigate).toBeCalledTimes(1);
      });
    });
  });
});
