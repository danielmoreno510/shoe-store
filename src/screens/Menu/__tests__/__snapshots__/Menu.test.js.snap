// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component GIVEN a set of valid props WHEN the component is rendered THEN should display a regular Menu 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <ScrollView>
    <View
      key="Home"
      style={
        Object {
          "paddingHorizontal": 16,
        }
      }
    >
      <ForwardRef
        onPress={[Function]}
        style={
          Object {
            "paddingVertical": 12,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Component
            name="home"
            type="MaterialIcons"
          />
          <Text
            style={
              Object {
                "color": "black",
                "fontSize": 16,
                "paddingHorizontal": 10,
              }
            }
          >
            Home
          </Text>
        </View>
      </ForwardRef>
    </View>
    <View
      key="Shoes List"
      style={
        Object {
          "paddingHorizontal": 16,
        }
      }
    >
      <ForwardRef
        onPress={[Function]}
        style={
          Object {
            "paddingVertical": 12,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Component
            name="shopping-bag"
            type="MaterialIcons"
          />
          <Text
            style={
              Object {
                "color": "black",
                "fontSize": 16,
                "paddingHorizontal": 10,
              }
            }
          >
            Shoes List
          </Text>
        </View>
      </ForwardRef>
    </View>
    <View
      key="Cart"
      style={
        Object {
          "paddingHorizontal": 16,
        }
      }
    >
      <ForwardRef
        onPress={[Function]}
        style={
          Object {
            "paddingVertical": 12,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Component
            name="shopping-cart"
            type="MaterialIcons"
          />
          <Text
            style={
              Object {
                "color": "black",
                "fontSize": 16,
                "paddingHorizontal": 10,
              }
            }
          >
            Cart
          </Text>
        </View>
      </ForwardRef>
    </View>
    <View
      key="About"
      style={
        Object {
          "paddingHorizontal": 16,
        }
      }
    >
      <ForwardRef
        onPress={[Function]}
        style={
          Object {
            "paddingVertical": 12,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Component
            name="info"
            type="MaterialIcons"
          />
          <Text
            style={
              Object {
                "color": "black",
                "fontSize": 16,
                "paddingHorizontal": 10,
              }
            }
          >
            About
          </Text>
        </View>
      </ForwardRef>
    </View>
  </ScrollView>
</ForwardRef(SafeAreaView)>
`;
